### ISSUE TITLE
Investigate flaky tests

### ISSUE THREAD

**Issue by andrewpillar** [djinn-ci/djinn#1]
**Version** `devel  +c65c3f5 Sat Apr 24 16:37:30 2021 +0100`

**Environment** [Hosted](https://djinn-ci.com)

**Describe the bug**
The `make.1` stage in the manifest for building Djinn is failing due to the test `Test_TagStoreCreate`. This has failed twice so far, but passed every other time. This appears to be happening sporadically, needs further investigation.

**Reproduction steps**
Hard to reproduce due to the sporadic nature. 

**Logs**
See the below snippet from the build output,

```
+ cd djinn
+ ./make.sh
go: downloading github.com/DATA-DOG/go-sqlmock v1.5.0
--- FAIL: Test_TagStoreCreate (0.00s)
    tag_test.go:120: build.(*TagStore).Create - build/tag.go:189: database.Store.Create - database/database.go:451: Query 'INSERT INTO build_tags (build_id, name, user_id) VALUES ($1, $2, $3) RETURNING id', arguments do not match: argument 0 expected [int64 - 3] does not match actual [int64 - 10]
FAIL
```

**Additional context**
The builds have been tagged as `flaky`, and can be seen here: https://djinn-ci.com/n/djinn-ci/djinn?tag=flaky, also listed below:

* https://djinn-ci.com/b/djinn-ci/5
* https://djinn-ci.com/b/djinn-ci/7

**Comment by andrewpillar** [comment]
So, the test is failing because a call to `WithArgs` expects the arguments given to the query in a certain order. During query construction the values are pulled from a map, which is unordered (as to be expected). So, on occasion the arguments will be in the correct order, and other times they won't be.

The call to `WithArgs` is as below,

    // build/tag_test.go
    mock.ExpectQuery(
        "^INSERT INTO build_tags (.+) VALUES (.+)$",
    ).WithArgs(3, 10, "tag1").WillReturnRows(
        sqlmock.NewRows([]string{"id"}).AddRow(1),
    )

and the query construction is,

    // database/database.go
    modelVals := m.Values()

    cols := make([]string, 0, len(modelVals))
    vals := make([]interface{}, 0, len(modelVals))

    for k, v := range modelVals {
        cols = append(cols, k)
        vals = append(vals, v)
    }

    q := query.Insert(
        table,
        query.Columns(cols...),
        query.Values(vals...),
        query.Returning("id"),
    )

**Comment by andrewpillar** [comment]
Put in a fix that I'm happy with, committed [8957eb6](https://github.com/djinn-ci/djinn/commit/8957eb6aca62326ba1b65890865aead59be8c529). The build for this is currently running through [here](https://djinn-ci.com/b/djinn-ci/14), so it should pass.
